### PR TITLE
Fix detection of blue-green process in cts scenarios

### PR DESCRIPTION
**Pull request by theghost5800**
JIRA:LMCROSSITXSADEPLOY-2362

### CODE REVIEW

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessTypeParser.java [hunk 1]
```diff
@@ -45,13 +40,30 @@ public ProcessType getProcessType(DelegateExecution execution, boolean required)
...
+        String processDefinitionId = execution.getProcessDefinitionId();
+        int serviceIdEndIndex = processDefinitionId.indexOf(':');
+
+        if (serviceIdEndIndex == -1) {
             String correlationId = VariableHandling.get(execution, Variables.CORRELATION_ID);
             throw new SLException(MessageFormat.format(Messages.UNKNWON_SERVICE_ID_FOR_PROCESS_0, correlationId));
         }
-        return serviceId;
+
+        String serviceId = processDefinitionId.substring(0, serviceIdEndIndex);
```
Comment (radoslav-d) [inline multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessTypeParser.java:52]: This code is confusing at first sight. Maybe extracting it in a method with a name like `getServiceIdFromProcessDefinition(String processDefinitionId)` will help readability.

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessTypeParser.java [hunk 2]
```diff
@@ -45,13 +40,30 @@ public ProcessType getProcessType(DelegateExecution execution, boolean required)
...
     }
 
-    protected String getServiceId(DelegateExecution execution) {
-        String serviceId = VariableHandling.get(execution, Variables.SERVICE_ID);
-        if (serviceId == null) {
+    public ProcessType getProcessTypeFromExecution(DelegateExecution execution) {
+        String processDefinitionId = execution.getProcessDefinitionId();
+        int serviceIdEndIndex = processDefinitionId.indexOf(':');
+
+        if (serviceIdEndIndex == -1) {
```
Comment (radoslav-d) [inline multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessTypeParser.java:47]: Is there a case where the process definition id has a different structure or invalid value at all?

**Review by radoslav-d — APPROVED** [review]
Is any adaptation needed in the xsa-multiapps-controller?

**Comment by theghost5800** [comment]
Fixed with https://github.com/cloudfoundry-incubator/multiapps-controller/pull/1050

**Pull request closed without merging**